### DISCUSSION TITLE
Config option for the amount of seconds between state messages

### DISCUSSION THREAD

**Discussion by sheilbronn** [tsightler/ring-mqtt#759] [Feature Requests]
Hallo!
Currently state messages are emitted every 300 seconds (5 minutes). 
However, this chattiness doesn't seem to be necessary, since the values hardly change. 
Would it be possible to **make the duration configurable (or to increase the duration to 15 minutes)?**

Thanks from a happy user (using ring-mqtt to integrate a Ring Intercom to a local OpenHab instance)

```
224550 ring/2516921f-.../intercom/343ea4b14e1c/battery/attributes {"batteryLevel":92}
224550 ring/2516921f-.../intercom/343ea4b14e1c/wireless/attributes {"wirelessNetwork":"TOEPFE","wirelessSignal":-49}
225052 ring/2516921f-.../intercom/343ea4b14e1c/info/state {"batteryLevel":92,"firmwareStatus":"Up to Date","lastUpdate":"2023-                       11-19T17:32:00Z","wirelessNetwork":"TOEPFE","wirelessSignal":-49}
225052 ring/2516921f-.../intercom/343ea4b14e1c/battery/attributes {"batteryLevel":92}
225052 ring/2516921f-.../intercom/343ea4b14e1c/wireless/attributes {"wirelessNetwork":"TOEPFE","wirelessSignal":-49}
225550 ring/2516921f-.../intercom/343ea4b14e1c/info/state {"batteryLevel":92,"firmwareStatus":"Up to Date","lastUpdate":"2023-                       11-19T17:32:00Z","wirelessNetwork":"TOEPFE","wirelessSignal":-49}
225550 ring/2516921f-.../intercom/343ea4b14e1c/battery/attributes {"batteryLevel":92}
225550 ring/2516921f-.../intercom/343ea4b14e1c/wireless/attributes {"wirelessNetwork":"TOEPFE","wirelessSignal":-49}
230050 ring/2516921f-.../intercom/343ea4b14e1c/info/state {"batteryLevel":92,"firmwareStatus":"Up to Date","lastUpdate":"2023-                       11-19T17:32:00Z","wirelessNetwork":"TOEPFE","wirelessSignal":-49}
```

**Comment by tsightler** [comment]
Attribute data is sent every 5 minutes, all other state data, such as sensor state changes, etc., are updated only when a change is detected.   The attribute update code is common across all devices, some of which have much more frequent changes, just because you have a device that doesn't change often doesn't mean other Ring devices don't.

I certainly agree it could be improved to only update on changes, etc, but that adds a lot of complication to compare state, and it's been this way for more than 5 years and no one has ever brought up a single functional issue with it.  What harm is this causing you and, other than "it's chatty" can you describe any functional or technical benefit of changing it because, to me, sending a few dozen bytes of data every 5 minutes to a service that can easily handle 10's of thousands of messages/second is no real issue.

**Reply by sheilbronn** [reply]
Hello, thanks for the fast reply!
I didn't know that there are many more attributes depending on the device. The ones I saw for the Ring Intercom, i.e. batterylevel and wireless data, seemed quite "constant" to me. 
I agree, it's not a real problem just the desire to limit messages on the MQTT bus as much as possible and sensible.

**Comment by sheilbronn** [comment]
One more question please:
Could you please point me to the place in the code where the 5 minutes are defined.
Thanks!
